rasa/nlu/featurizers/sparse_featurizer/lexical_syntactic_featurizer.py

```diff
@@ -69,82 +85,269 @@ def required_components(cls) -> List[Type[Component]]:
+            return token_position == 0
+        return cls._FUNCTION_DICT[feature_name](token)
+
+    @staticmethod
+    def get_default_config() -> Dict[Text, Any]:
+        """Returns the component's default config."""
+        return {
+            **SparseFeaturizer2.get_default_config(),
+            LexicalSyntacticFeaturizerGraphComponent.FEATURES: [
+                ["low", "title", "upper"],
+                ["BOS", "EOS", "low", "upper", "title", "digit"],
+                ["low", "title", "upper"],
+            ],
+        }
+
     def __init__(
         self,
-        component_config: Dict[Text, Any],
-        feature_to_idx_dict: Optional[Dict[Text, Any]] = None,
-    ):
-        super().__init__(component_config)
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
+    ) -> None:
+        """Instantiates a new `LexicalSyntacticFeaturizer` instance."""
+        super().__init__(config)
+        # graph component
+        self._model_storage = model_storage
+        self._resource = resource
+        self._execution_context = execution_context
+        # featurizer specific
+        self._feature_config = self._config.get(self.FEATURES, [])
+        self._feature_to_idx_dict = {}
+        self._number_of_features = -1
+        # where the negative value indicates that feature_to_idx hasn't been set yet
+
+    def validate_config(cls, config: Dict[Text, Any]) -> None:
+        """Validates that the component is configured properly."""
+        SparseFeaturizer2.validate_config(config)
+        feature_config = config.get(cls.FEATURES, [])
+        if not feature_config:
+            rasa.shared.utils.io.raise_warning(
+                f"Expected 'features' to be configured for the " f"{cls.__name__} ",
+                doc=DOCS_URL_COMPONENTS,
+            )
+            feature_config = []  # in case it was None
+        message = (
+            f"Expected configuration of `features` to be a list of lists that "
+            f"that contain names of lexical and syntactic features "
+            f" (i.e. {cls.SUPPORTED_FEATURES})."
+            f"Received {feature_config} instead. "
+        )
+        try:
+            configured_feature_names = set(
+                feature_name
+                for pos_config in feature_config
+                for feature_name in pos_config
+            )
+        except TypeError as e:
+            raise InvalidConfigException(message) from e
+        if configured_feature_names.difference(cls.SUPPORTED_FEATURES):
+            raise InvalidConfigException(message)
+
+    def validate_compatibility_with_tokenizer(self, tokenizer: Tokenizer) -> None:
+        """Validate a configuration for this component in the context of a recipe."""
+        # NOTE: this wasn't done before, there was just a comment
+        # TODO: add (something like) this to graph validation
```

Comment:
I think if we do this than we should do it in the `Recipe`. The graph schema validation doesn't know / doesn't care about the types of the components.

Reply:
Sounds good - I didn't think much about Recipe vs graph schema :)